netutils/netinit/netinit.c

```diff
@@ -343,6 +353,86 @@ static void netinit_set_macaddr(void)
 #  define netinit_set_macaddr()
 #endif
 
+#ifdef CONFIG_FSUTILS_IPCFG
+/****************************************************************************
+ * Name: netinit_get_ipaddrs
+ *
+ * Description:
+ *   Setup IP addresses.
+ *
+ *   For 6LoWPAN, the IP address derives from the MAC address.  Setting it
+ *   to any user provided value is asking for trouble.
+ *
+ ****************************************************************************/
+
+static int netinit_get_ipaddrs(void)
+{
+#ifdef CONFIG_NETINIT_DHCPC
+  bool use_dhcp   = false;
+#endif
+  bool use_static = false;
+#ifdef CONFIG_NET_IPv4
+  struct in_addr addr;
+  int ret;
+
+  ret = ipcfg_read(NET_DEVNAME, (FAR struct ipcfg_s *) &g_netconf, AF_INET);
+  if (ret < 0)
+    {
+      return ret;
+    }
+  else
+    {
+      use_static = g_netconf.proto & BOOTPROTO_STATIC;
+
+#ifdef CONFIG_NETINIT_DHCPC
+      use_dhcp = g_netconf.proto & IPv4PROTO_DHCP;
+      addr.s_addr = use_dhcp ? 0 : HTONL(g_netconf.ipaddr);
+#else
+      addr.s_addr = HTONL(g_netconf.ipaddr);
+#endif
```

Comment:
Ditto.  And for all over places where HTONL is used on addresses returned by ipcfg.

Reply:
"